tests/fault_tolerance/deploy/client.py

```diff
@@ -423,20 +442,77 @@ def run_aiperf(
             logger.error(f"Error in attempt {attempt + 1}: {str(e)}")
             all_results.append({"attempt": attempt + 1, "error": str(e)})
 
-        # Sleep before next attempt (if not the last attempt)
-        if not success and attempt < max_attempts - 1:
+        # Sleep before next attempt (if not the last attempt and not continuous load)
+        if not success and attempt < max_attempts - 1 and not continuous_load:
             time.sleep(retry_delay)
 
-    if success:
+    if success and not continuous_load:
         logger.info(
             f"AI-Perf successfully completed all {requests_per_client} requests for {pod_name}"
         )
+    elif success and continuous_load:
+        logger.info(
+            f"AI-Perf sustained continuous load for {pod_name} and existed succesfully"
+        )
     else:
         logger.error(f"AI-Perf failed all {max_attempts} attempts for {pod_name}")
 
     return success
 
 
+def run_aiperf_with_signal_handling(
+    cmd_attempt: List[str],
+    logger: logging.Logger,
+    timeout: int,
+) -> subprocess.CompletedProcess:
+    """
+    Run aiperf with signal handling for graceful shutdown.
+
+    Handles SIGINT forwarding and timeout when running with subprocess.Popen.
+    This ensures that Ctrl-C and SIGINT are properly forwarded to the subprocess
+    so it can clean up gracefully and write results files.
+    """
+    proc = subprocess.Popen(
+        cmd_attempt,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+        text=True,
+        stdin=subprocess.DEVNULL,
+    )
+
+    # Set up signal handler to forward SIGINT to subprocess
+    def signal_handler(signum, frame):
+        logger.info(f"Received signal {signum}, forwarding to aiperf subprocess")
```

Comment:
question: we are always forwarding sigint and not necessarily the signal we recieved - is that intended?